src/Core/Services/MetadataProviders/MetadataProviderFactory.cs

```diff
@@ -56,13 +56,24 @@ private void ConfigureMetadataProviders()
                     _ => throw new NotSupportedException(dataSource.DatabaseTypeNotSupportedMessage),
                 };
 
-                _metadataProviders.Add(dataSourceName, metadataProvider);
+                // In a hot reload scenario, the metadata might be updated dynamically. The below code will ensure that:
```

Comment:
Can remove this and instead just create a new set of metadataProviders so we just build clean.